dan_layer/wallet/storage_sqlite/src/writer.rs

```diff
@@ -455,21 +455,23 @@ impl WalletStoreWriter for WriteTransaction<'_> {
             .select(accounts::id)
             .filter(accounts::address.eq(vault.account_address.to_string()))
             .first::<i32>(self.connection())
+            .optional()
             .map_err(|e| WalletStorageError::general("vaults_insert", e))?;
 
-        let values = (
-            vaults::account_id.eq(account_id),
-            vaults::address.eq(vault.address.to_string()),
-            vaults::balance.eq(vault.balance.value()),
-            vaults::resource_address.eq(vault.resource_address.to_string()),
-            vaults::resource_type.eq(format!("{:?}", vault.resource_type)),
-            vaults::token_symbol.eq(vault.token_symbol),
-        );
-        diesel::insert_into(vaults::table)
-            .values(values)
-            .execute(self.connection())
-            .map_err(|e| WalletStorageError::general("vaults_insert", e))?;
-
+        if let Some(account_id) = account_id {
```

Comment:
I think that we should return an error here if account_id is None because it should be invalid to call `vault_insert` without an associated account for the vault. The `has_vault` check above should prevent this error.